Rewrite Peak Watcher to support multiple level types and watching things other than tracks.

jcsteh opened the pull request:
Instead of choosing two tracks within the dialog, the currently focused track is used. You are then asked to choose which watched value you want to configure, either value 1 or value 2. If either of these values has been configured already, the menu provides information about the existing configuration as a reminder. This allows you to use separate level types for the two watched values.

In addition, Peak Watcher now supports gain reduction on track FX which support it. To use this, open the Peak Watcher while focused on an effect in the FX chain dialog. Support for take FX is possible, but not implemented yet.

This is going to need a fair bit of testing. Also, I worry the UX might not be entirely intuitive, but I honestly couldn't come up with anything better that allows us to support all the things we want to support. I'm hoping we can clarify the UX here if necessary with terminology changes alone. Things to consider:

1. I'm concerned that users might not understand what they're being asked to choose when asked to select a value to configure via the context menu that opens before the dialog appears.
2. Rather than saying "track x" or "first track", the Peak Watcher now talks in terms of watched values; value 1 channel 1, etc. This is necessary both to support things that aren't tracks and also to avoid the need to suffix the level type (peak dB, momentary LUFS, etc.) if the user happens to be watching two different values on the same track.

I haven't updated the readme yet. I'll do that once we solidify the UX.

Fixes #617. Supersedes #508.

jcsteh commented:
Instead of "value 1" and "value 2", I considered "watcher 1" and "watcher 2". However, I wasn't sure if that made things clearer or not, so I went with "value".

jcsteh commented:
Yup, I still hate having to support Mac.

jcsteh commented:
If this doesn't work, I swear I'm going to make this repository Windows only and someone else can maintain Mac in a separate fork.

Okay, I'm not going to do that, but I really wish I could.

Lo-lo78 commented:
@jcsteh Nice work, I love how you organized View Peak Watcher.
A problem is that if the audio track is in play, it always reads the value without stopping at the entered value of the When level reaches.
When I pause the volume, for example, it goes back to -150 db. The last value does not remain.
The maximum peak should no longer be updated, in this way we can understand when the maximum point of the db has arrived, in this Build by OSARA it always reads without stopping at the maximum peak; in pause it updates by resetting to -150 db.
Greetings

jcsteh commented:
I can't reproduce this. Can you provide the exact settings you used? Did
you set Hold highest level until reset?

Lo-lo78 commented:
Message ID: ***@***.***>

Hi James,

sorry, I took the test wrong, everything is perfect.

It is OK!

My mistake was entering a value higher than 0 in the When level reaches!

Sorry James for my mistake and thank you very much!

👍

jcsteh commented:
I'll have to try that. You should be able to enter whatever value you like in "when level reaches". It should be independent of the "hold highest level" setting unless I broke something.

ScottChesworth commented:
Christmas literally came early, this is gonna be super powerful!

> I'm concerned that users might not understand what they're being asked to choose when asked to select a value to configure via the context menu that opens before the dialog appears.
> 
Perhaps Primary and Secondary instead of value 1 and 2? Those terms would translate universally, and I think maybe there's some intuitiveness to be had from more distinction between words and numbers in reports. Words for what's being watched, numbers are the data. I'm explaining it badly but hopefully you get what I mean.

> I'm hoping we can clarify the UX here if necessary with terminology changes alone.
> 
Sure, a few thoughts to run by you on that front:
1. I don't think "Current:" needs to be there in the menu summary. I copied a few configurations, tried removing it, and everything still made sense, just was quicker to hear.
2. In the popup menu, I'm concerned that "disabled" could be confused with a dimmed menu option. Can we try "not set yet" or "not configured" instead?
3. The summary reports in the popup menu use track numbers right now. Any chance of names instead, reverting to numbers if a track that's being followed is unnamed?
4. Can you strip plug-in vendor from the summary report when gain reduction is being watched?
5. Dunno whether "Follow current track" is the clearest label for the default check box. Maybe "Move watcher with track selection", "Watcher tracks selection", "Watcher mirrors track selection"?
On this, I guess I'm wondering why you've moved away from choosing a track from a list box with the "follow current" option being topmost? Could be habitual, but I think perhaps that was a more intuitive UX.

Flameborn commented:
> Perhaps Primary and Secondary instead of value 1 and 2? Those terms would translate universally, and I think maybe there's some intuitiveness to be had from more distinction between words and numbers in reports. Words for what's being watched, numbers are the data. I'm explaining it badly but hopefully you get what I mean.

Reaper has this with its external editors, so in this sense it's more universal, I agree. On the other hand, the terms *primary* and *secondary* suggest to me that there's a relationship between these, i.e. one is better than the other, which is not true. I would just simply leave these as *Level 1* and *Level 2*.

jcsteh commented:
> Perhaps Primary and Secondary instead of value 1 and 2? Those terms would translate universally, and I think maybe there's some intuitiveness to be had from more distinction between words and numbers in reports.

It'd still have to be primary and secondary "something"; primary value, primary watcher, etc. Sure, we can shorten to just "primary" when reporting, but not in the documentation nor in the description of the actions.

I too worry that this suggests that primary is somehow more important or better than secondary. I guess we could say "first value" and "second value" though? However, then there's still potential confusion with the term "value".

>     1. I don't think "Current:" needs to be there in the menu summary. I copied a few configurations, tried removing it, and everything still made sense, just was quicker to hear.

The reason I added this is that I didn't want the user to think that they were being told what they were about to set, but rather, that this is what is currently set. That is, what they're hearing there isn't the setting they'll make when they press enter.

>     2. In the popup menu, I'm concerned that "disabled" could be confused with a dimmed menu option. Can we try "not set yet" or "not configured" instead?

Sure.

>     3. The summary reports in the popup menu use track numbers right now. Any chance of names instead, reverting to numbers if a track that's being followed is unnamed?

I can do that, but:

- It wouldn't be clear that the user is dealing with a track. Right now, there is a "track" prefix to make that clear. I guess you could say "track main vocal", but that feels weird. But maybe it's okay.
- It will make the report for gain reduction (which includes the FX name) even longer than it already is.

>     4. Can you strip plug-in vendor from the summary report when gain reduction is being watched?

I can, but it'll require a tiny bit of shifting things around. I may (or may not) leave that for a follow-up.

>     5. Dunno whether "Follow current track" is the clearest label for the default check box.

That's what we called the combo box option previously. Why is that any less clear here?
>        On this, I guess I'm wondering why you've moved away from choosing a track from a list box with the "follow current" option being topmost? Could be habitual, but I think perhaps that was a more intuitive UX.

Either I'm missing something or you're forgetting one of the reasons we're doing this in the first place. :) We need the Peak Watcher target to be based on the focus. Otherwise, it's not possible to support values on things other than tracks; e.g. gain reduction on a track effect (and eventually a take effect). Trying to show track effects, items, takes, etc. in this dialog is going to get nasty. I thought we'd been over all of this in #618. If I'd known this was still a sticking point, I might not have spent 10+ hours on implementing it based on focus.

jcsteh commented:
One problem with primary/first/secondary/second is that the menu items couldn't have shortcuts of "1" and "2", which feels more intuitive than p/f/s. I guess we could name the menu items "1: primary value, blah", but that creates redundant verbosity. Maybe we can live with p/f/s as shortcuts though.

jcsteh commented:
Although we could do 1st, 2nd, etc., which would allow for 1 and 2 as shortcut keys.

jcsteh commented:
>    On this, I guess I'm wondering why you've moved away from choosing a track from a list box with the "follow current" option being topmost? Could be habitual, but I think perhaps that was a more intuitive UX.

To clarify, I think this probably was a more intuitive UX. Unfortunately, it doesn't allow us to support the things we want to support, namely gain reduction and the like.

Flameborn commented:
> I guess we could say "first value" and "second value" though? However, then there's still potential confusion with the term "value".

If the value applies to one thing only, in this case to volume levels, then I think it would be better to specify what value we are asking for, hence my suggestion of Level 1 and Level 2, which I guess we could change to Peak Level 1 and Peak Level 2. This way, there is no relationship between the two watched values, and we also describe what needs to be specified, while we leave the 1 and 2 hotkeys as is in the menus.

> The reason I added this is that I didn't want the user to think that they were being told what they were about to set, but rather, that this is what is currently set. That is, what they're hearing there isn't the setting they'll make when they press enter.

I think we can address this if we use something like *Set Peak level 1* as the menu item text when value 1 is not configured, and we can remove the word *current* when it's set. We could also change the menu item to be checked, or just put the word *set* afterwards to make it clear that this value is already configured.

> * It wouldn't be clear that the user is dealing with a track. Right now, there is a "track" prefix to make that clear. I guess you could say "track main vocal", but that feels weird. But maybe it's okay.

I think if you have 200 plus tracks in a project, which in my case can happen quite easily, track names are a lot more helpful, as we tend to remember names more reliably than numbers in general.

> * It will make the report for gain reduction (which includes the FX name) even longer than it already is.

I think in this case it's a worthy sacrifice.

> We need the Peak Watcher target to be based on the focus. Otherwise, it's not possible to support values on things other than tracks; e.g. gain reduction on a track effect (and eventually a take effect). Trying to show track effects, items, takes, etc. in this dialog is going to get nasty.

I agree, using the focus as a reference makes things a lot easier, and it saves an incredible amount of time as well. A lot of other OSARA features use the focus, so this should be second nature to users.

jcsteh commented:
> If the value applies to one thing only, in this case to volume levels, then I think it would be better to specify what value we are asking for, hence my suggestion of Level 1 and Level 2, which I guess we could change to Peak Level 1 and Peak Level 2.

While peak can indeed have a more general meaning, my understanding is that audio folks tend to think of waveform peak (peak dB) when they read "peak". In contrast, Peak Watcher supports many different types of levels; LUFS aren't really peak, for example. My concern is that users would confuse these things. I'm not convinced "level" is any better than "value" though, but I'm happy to change to that if there's consensus that it's more intuitive.

> I think we can address this if we use something like _Set Peak level 1_ as the menu item text when value 1 is not configured, and we can remove the word _current_ when it's set.

I don't really see how this makes it clear that the info at the end of the menu item refers to the existing setting, not the current focus. And removing that info altogether would mean the user has no way of checking what they previously configured, since it was based on the focus at the time they configured it.

> I think if you have 200 plus tracks in a project, which in my case can happen quite easily, track names are a lot more helpful, as we tend to remember names more reliably than numbers in general.

Yeah, I tend to agree. I was just concerned about verbosity here, but I think I agree it's worth the sacrifice in this case.

jcsteh commented:
>     1. I don't think "Current:" needs to be there in the menu summary.

I dropped this despite my concerns, since it seems both of you aren't concerned this is going to be less intuitive without it.

>     2. In the popup menu, I'm concerned that "disabled" could be confused with a dimmed menu option.

Changed to "not set".

>     3. The summary reports in the popup menu use track numbers right now. Any chance of names instead, reverting to numbers if a track that's being followed is unnamed?

Done.

>     4. Can you strip plug-in vendor from the summary report when gain reduction is being watched?

Done.

Flameborn commented:
> While peak can indeed have a more general meaning, my understanding is that audio folks tend to think of waveform peak (peak dB) when they read "peak". In contrast, Peak Watcher supports many different types of levels; LUFS aren't really peak, for example. My concern is that users would confuse these things. I'm not convinced "level" is any better than "value" though, but I'm happy to change to that if there's consensus that it's more intuitive.

Ah, I forgot about that, LUFS level does not make much sense indeed.

> I don't really see how this makes it clear that the info at the end of the menu item refers to the existing setting, not the current focus. And removing that info altogether would mean the user has no way of checking what they previously configured, since it was based on the focus at the time they configured it.

I think if the menu item text says something like *-16 set* or *-16 checked*, rather than *Set value 1*, this should be a clear indication that we are working with an existing value. So in this sense it's just a status indication only, the same way the word *current* indicates that this value is the currently set one, with the exception that the word *set* would be at the end of the menu item, so one gets to the actual value quicker. This is why @ScottChesworth suggested to remove the word *current* as well.

> Yeah, I tend to agree. I was just concerned about verbosity here, but I think I agree it's worth the sacrifice in this case.

No worries, we sometimes tend to overthink things, because achieving a healthy balance that does not result in an information overload can be tricky sometimes. Happens to me all the time 😀

jcsteh commented:
> I think if the menu item text says something like _-16 set_ or _-16 checked_, rather than _Set value 1_, this should be a clear indication that we are working with an existing value.

It doesn't say "-16" (or whatever numeric value), though, and I don't think it should. First, that would make it difficult to figure out which value (first or second) you're adjusting. Second, nothing you do here changes the actual numeric value reported; you're just changing what the Peak Watcher is watching. In full sentences, what we're trying to say is "Configure the first value the Peak Watcher will watch. It's currently watching the track main vocal, specifically the peak dB level."

If you want to actually report the numeric value, you do that the same way you always did: with alt+f11, etc.

Flameborn commented:
Looks good here.

> I dropped this despite my concerns, since it seems both of you aren't concerned this is going to be less intuitive without it.

I suggested the word *set* after the value to make it more intuitive, but I think this is fine as well, since if values are not configured, the menu item will not have a set value shown.

Flameborn commented:
> It doesn't say "-16" (or whatever numeric value), though, and I don't think it should. First, that would make it difficult to figure out which value (first or second) you're adjusting. Second, nothing you do here changes the actual numeric value reported; you're just changing what the Peak Watcher is watching. In full sentences, what we're trying to say is "Configure the first value the Peak Watcher will watch. It's currently watching the track main vocal, specifically the peak dB level."

I was actually referring to the value you set, rather than the current level, which you are right, you can report with the appropriate hotkeys. But it looks good.

I'm currently getting unicode characters as opposed to a track name in the watcher menu at the moment, using the Mac universal binary of Reaper.

ScottChesworth commented:
> I guess we could say "first value" and "second value" though? However, then there's still potential confusion with the term "value".
> 
I like your idea of 1st watcher, 2nd watcher. That's clear, concise, and we can keep the 1 and 2 shortcuts.

> > 5. Dunno whether "Follow current track" is the clearest label for the default check box.
> > ```
> 
> That's what we called the combo box option previously. Why is that any less clear here?
> 
To me, follow current track as an option in proximity to other track names I'll recognize from my project was clearer, because it's obvious that it's an either or type dealio. Like, I can either choose to follow the current track as selection changes, or lock it to one of these others. I get why the list of tracks has gone away (more on that in a sec), but still think the two behaviours could be made easier to understand.
How about two radio buttons instead of a check box? Default would be "Follow <track name or number>", other option would be "follow track selection". 

> > ```
> >    On this, I guess I'm wondering why you've moved away from choosing a track from a list box with the "follow current" option being topmost? Could be habitual, but I think perhaps that was a more intuitive UX.
> > ```
> 
> Either I'm missing something or you're forgetting one of the reasons we're doing this in the first place. :)
> 
You're not missing anything, I'm back at the grindstone before I should be. Still kinda ill and probably semi-scatty here. Sorry for the confusion.

ScottChesworth commented:
> > 3. The summary reports in the popup menu use track numbers right now. Any chance of names instead, reverting to numbers if a track that's being followed is unnamed?
> > ```
> 
> Done.
> 
Thanks. Last tweak, can we get a comma after the track name or number? It adds a tiny bit of extra time to hear the report in full, but that extra separation also adds a ton of clarity IMO.

jcsteh commented:
Hmm. I guess I kinda follow the follow confusion. I think the two radio buttons introduce some other confusion, though: now the term follow is used for both cases, where I don't really think of watching a specific track as "following".

The other thing to note is that "track selection" is incorrect. It uses the last touched track (AKA current track), which is not necessarily the same as track selection.

What about labelling the checkbox "Follow when current track changes)?

I can add the comma after the track name/number, sure. I'm going away for a week or so now, so there won't be any code updates on this until I get back.

Finally, I realised my comment about your forgetting the point came off as kinda snarky. Sorry about that; this stuff is complicated. I was being a bit touchy given the hours I sank into this, but it was uncalled for.

ScottChesworth commented:
> What about labelling the checkbox "Follow when current track changes)?
> 
Ooh, that feels closer yeah. How about "Follow when last-touched track changes", or just "Watch last-touched track"?

> I can add the comma after the track name/number, sure. I'm going away for a week or so now, so there won't be any code updates on this until I get back.
> 
I can probably do that from here. It'll be fun trying anyway.

> Finally, I realised my comment about your forgetting the point came off as kinda snarky. Sorry about that; this stuff is complicated. I was being a bit touchy given the hours I sank into this, but it was uncalled for.
> 
No worries man, historically, it takes many years of festering grudges to get me to leave a band here. :)

jcsteh commented:
I think I prefer "Follow when last track changes". My concern with just "Follow last touched track" is that it has pretty much the same problem as "Follow current track" in that the user might think it just means to use the last touched track as it is now, not realising this is about dynamically following the track as the track changes.

jcsteh commented:
While we're talking about "1st watcher", I wonder whether we should also say "1st chan". That way, we can report "1st watcher 1st chan -5.6", dropping the word "level" and thus reducing verbosity. We can't drop "level" with "chan 1" because it'd be confused with the value; "chan 1 -5.6".

jcsteh commented:
Okay. I think I've implemented all the proposed changes here. Let me know what you think.

ScottChesworth commented:
All looking rather marvelous here language-wise! This rulez (with a freakin' z).
You're gonna hate how long it's taken me to notice this, but using gain reduction in situ just now, there's a logic floor in the default settings. Peak Watcher is holding onto the highest setting until reset, but in context of gain reduction, that means the minimum amount of GR gets held.

jcsteh commented:
Yeah, I was wondering whether we'd hit that minimum/maximum issue with certain level types. This probably also applies to "Notify when level reaches", which currently notifies for levels greater than the specified number. Is that undesirable too? Should it be levels less than the specified number?

I also wonder about loudness range. Do we want to hold the max there or the min? And what about the notify level: greater than or less than?

I'm not sure how to represent this in the UI. "until level reaches" is okay, I think, sincereaches could be either less or greater. But what about "Hold highest level"? We'd need a term there which somehow suggests highest or lowest depending on the level type.

jcsteh commented:
Wait a sec. Why would this be holding the minimum for gain reduction? I haven't actually played with this, but given the term "reduction", I would have expected a positive number, since we're saying the compressor is reducing by that amount. Or does it actually use a negative number? In that case, it should really be called gain impact or something, but blurp...

ScottChesworth commented:
Yep, gain reduction is a negative scale, so -10 is more GR than -5. I know, doesn't make any sense to me either.
Does the labeling need to be explicit about whether we're holding highest or lowest value? I mean, so long as we're sure we are actually holding the right end of the range to be useful for each of the options we're providing, then wouldn't just a generic "Hold until reset" do the trick?

jcsteh commented:
Just to clarify, for gain reduction, should auto notifications definitely occur when the level is less than the specified value? That is, would a user always want to automatically know when the level is less than, not greater than?

ScottChesworth commented:
> would a user always want to automatically know when the level is less than, not greater than?
> 
Yeah, I think so. Can't see how the inverse would be more useful in situ.

ScottChesworth commented:
Another idea for simplifying language, what if we change the "When level reaches" label to "Start reporting at". That way, the label can stay the same whether we're reporting above or below.

jcsteh commented:
> using gain reduction in situ just now, there's a logic floor in the default settings. Peak Watcher is holding onto the highest setting until reset, but in context of gain reduction, that means the minimum amount of GR gets held.

This should be fixed now.

ScottChesworth commented:
Thanks, gain reduction just worked a treat here in my first round of testing. Turns out I'm horrible at setting a compressor based on numbers alone at this stage, but that's not OSARA's fault. :)

ScottChesworth commented:
Hey @jcsteh, any thoughts on the Unicode characters that are appearing on Mac in the new context menu? My regular Mac tester was still getting those in the build that went up earlier today. Haven't had her check this most recent one yet but I didn't spot any activity around the Unicode stuff.

jcsteh commented:
I don't really know what's meant by "Unicode characters". Technically, any character is a Unicode character. :) I assume users are seeing garbage, but I'd really need to see an example of what they're seeing to have any clue. Is it only the track name (as suggested in https://github.com/jcsteh/osara/pull/623#issuecomment-1001174413) or does it also impact other content? What about when the track doesn't have a name?

jcsteh commented:
I can't see anything in the code that should be causing this. :(

ScottChesworth commented:
> I'd really need to see an example of what they're seeing to have any clue.
> 
I'll fire up the Mac later today, get some speech log for examination.

pitermach commented:
I happened to see this and had a quick look. The menu items appear as garbage characters. I'm seeing things like: @¶T, ³°T, â»T. The garbage characters aren't consistent either - if I command-tab out of reaper and back I get different chars every time and sometimes you just get completely blank items.

jcsteh commented:
Thanks @pitermach. That was very helpful. So the entire string is garbage, not just the track name.

I've hopefully fixed this now. Would you mind checking?

TimNoonanVoice commented:
Hi James,

I'm not getting any obvious garbage text here in watcher menus on MacOS 12.1.

However, I think I deleted the email with the problem version, so feedback by others is still needed.

Tim


On 5 Jan 2022, at 11:09 am, James Teh ***@***.***> wrote:


Thanks @pitermach <https://github.com/pitermach>. That was very helpful. So the entire string is garbage, not just the track name.

I've hopefully fixed this now. Would you mind checking?

—
Reply to this email directly, view it on GitHub <https://github.com/jcsteh/osara/pull/623#issuecomment-1005268070>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGNKY26U6AMFLV7YTLZMADDUUOD47ANCNFSM5KWSBPRQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
You are receiving this because you are subscribed to this thread.

ScottChesworth commented:
Jen just confirmed the fix is working on Monterey. Thanks man!

pitermach commented:
Also working great here now.

jcsteh commented:
Okay. I've updated the readme to document the new Peak Watcher.

@ScottChesworth , how would you like to proceed from here? Given the huge changes here and the potential risks of breakage, do you want to test it a bit more yourself and/or pull others in to test it more before we merge? Or are you ready for this to be merged now?

Lo-lo78 commented:
Hello everyone, happy 2022!

I have tested and in my Windows 10 system it works perfectly!

Congratulations James!

We await more feedback ...

👍

Message ID: ***@***.***>

ScottChesworth commented:
> are you ready for this to be merged now?
> 
I'd like to get at least one audio demo ready before merging, and do more testing while that's in progress. On the tail end of a very Covidy Christmas here, not talking or demoing much on groups yet because I'm still coughing every other word, but reckon I'll be able to record something next week.
In the meantime, tagging a few teachy people who'll want to start getting comfy with the new UX and features: @CastleStarter, @Justinmac84, @GarthHum.
I'll send a link to others who don't have accounts on GitHub.

ScottChesworth commented:
Are those latest commits to update terminology in the commit summaries @jcsteh, or did I miss some other more significant change?

jcsteh commented:
They were just slight tweaks to the access keys in the dialog. I realised
alt+d was Disable for hold but also the new Disable button. The new Disable
button is now alt+i, which is not ideal but is better than a conflicting
access key.

TimNoonanVoice commented:
Not sure if this is intentional, but there is an & at the start of the Follow Track element on Mac:

&Follow when last touched track changes unticked checkbox

If I'm supposed to report such things in another way, please let me know.

Thanks
Tim


On 8 Jan 2022, at 6:12 pm, James Teh ***@***.***> wrote:


They were just slight tweaks to the access keys in the dialog. I realised
alt+d was Disable for hold but also the new Disable button. The new Disable
button is now alt+i, which is not ideal but is better than a conflicting
access key.
—
Reply to this email directly, view it on GitHub <https://github.com/jcsteh/osara/pull/623#issuecomment-1007900220>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGNKY24BLV7QBGA4H4Y6WWLUU7PUXANCNFSM5KWSBPRQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
You are receiving this because you commented.

ScottChesworth commented:
Good catch, I'd missed that because of verbosity settings. Thanks man.

jcsteh commented:
Curious. The "&" is used to specify an access key and it should get handled
accordingly. But... if this is broken for the follow check box, are you
also seeing a spurious & character for other check boxes/radio buttons in
that dialog? Those are handled the same way, so I'm unsure why this would
be any different.

TimNoonanVoice commented:
Hi James,

No, the other buttons such as Ok, Cancel and Disable don't show the leading & character.

I'm running your latest watcher build, on MacOS 12.1. but I noticed the & in earlier builds too. 

Tim


On 10 Jan 2022, at 2:27 pm, James Teh ***@***.***> wrote:


Curious. The "&" is used to specify an access key and it should get handled
accordingly. But... if this is broken for the follow check box, are you
also seeing a spurious & character for other check boxes/radio buttons in
that dialog? Those are handled the same way, so I'm unsure why this would
be any different.
—
Reply to this email directly, view it on GitHub <https://github.com/jcsteh/osara/pull/623#issuecomment-1008515705>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGNKY26VZDFBCZ23GV3BHMTUVJG3TANCNFSM5KWSBPRQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
You are receiving this because you commented.

TimNoonanVoice commented:
Can't seem to find option to Nominate track to be watched on Mac!

Hi again James,

Feeling a bit un-thorough here, but on MacOS 12.1 , with all of the PR OSARA new watcher versions I've tried, after I enable watcher 1 or 2, there seems to be no way to nominate the track for watching. I missed this, because I was following current track.

Could this be related to the leading & problem - which is rather minor if track nomination is not showing up.

Any other Mac users who can test and confirm this on their set-ups?

thanks
Tim

On 10 Jan 2022, at 2:27 pm, James Teh ***@***.***> wrote:


Curious. The "&" is used to specify an access key and it should get handled
accordingly. But... if this is broken for the follow check box, are you
also seeing a spurious & character for other check boxes/radio buttons in
that dialog? Those are handled the same way, so I'm unsure why this would
be any different.
—
Reply to this email directly, view it on GitHub <https://github.com/jcsteh/osara/pull/623#issuecomment-1008515705>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGNKY26VZDFBCZ23GV3BHMTUVJG3TANCNFSM5KWSBPRQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
You are receiving this because you commented.

jcsteh commented:
As noted in the PR description, the track to be watched is now determined
by the focus at the time you run the Peak Watcher action. This was
necessary in order to support monitoring things other than tracks and to
allow multiple tracks to be configured completely independently.

TimNoonanVoice commented:
Ah,

Sorry. I read the discussions, but obviously didn't retain that point.

I wonder then if the interface should implicitly report the track it is watching, ;in case other people switching over to it haven't fully realise the new approach.

Tim

On 12 Jan 2022, at 4:31 pm, James Teh ***@***.***> wrote:


As noted in the PR description, the track to be watched is now determined
by the focus at the time you run the Peak Watcher action. This was
necessary in order to support monitoring things other than tracks and to
allow multiple tracks to be configured completely independently.
—
Reply to this email directly, view it on GitHub <https://github.com/jcsteh/osara/pull/623#issuecomment-1010652629>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGNKY25W33FMHP2XTQCLJOLUVUG4RANCNFSM5KWSBPRQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
You are receiving this because you commented.

TimNoonanVoice commented:
Ok,

I'm going to stop here. Realise now that when you enter the watcher a second time, this is presented.

Tim


On 12 Jan 2022, at 4:31 pm, James Teh ***@***.***> wrote:


As noted in the PR description, the track to be watched is now determined
by the focus at the time you run the Peak Watcher action. This was
necessary in order to support monitoring things other than tracks and to
allow multiple tracks to be configured completely independently.
—
Reply to this email directly, view it on GitHub <https://github.com/jcsteh/osara/pull/623#issuecomment-1010652629>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGNKY25W33FMHP2XTQCLJOLUVUG4RANCNFSM5KWSBPRQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
You are receiving this because you commented.

jcsteh commented:
When you enter the watcher menu, it reports whatever the watcher is
currently monitoring for your information. However, if you press enter to
reconfigure that watcher, it will use the focused track. As I noted in some
earlier comments, this was my main concern about the intuitiveness of this
user experience.

I could potentially have the Watcher dialog report what track it's
configuring (e.g. in the title of the dialog), but I worry that would cause
a lot of verbosity.

TimNoonanVoice commented:
I wonder, then, could the "not configured" message also state what happens if enter is pressed. Or, when it is pressed, could there be an announcement confirming what track is now being watched?

Otherwise, you only ever get advised of the watched track if you go in again to configure the second track; which some people may not do at the outset.

I'm only throwing out ideas here to reduce the likelihood of others missing what is happening, during their first experiences when encountering the new and changed ui.

Thanks
Tim

On 12 Jan 2022, at 5:31 pm, James Teh ***@***.***> wrote:


When you enter the watcher menu, it reports whatever the watcher is
currently monitoring for your information. However, if you press enter to
reconfigure that watcher, it will use the focused track. As I noted in some
earlier comments, this was my main concern about the intuitiveness of this
user experience.

I could potentially have the Watcher dialog report what track it's
configuring (e.g. in the title of the dialog), but I worry that would cause
a lot of verbosity.
—
Reply to this email directly, view it on GitHub <https://github.com/jcsteh/osara/pull/623#issuecomment-1010684439>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGNKY23MPKJUCZ6KS7D5LLDUVUN4HANCNFSM5KWSBPRQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
You are receiving this because you commented.

jcsteh commented:
> there is an & at the start of the Follow Track element on Mac: &Follow when last touched track changes unticked checkbox

Is this fixed now?

> Or, when it is pressed, could there be an announcement confirming what track is now being watched?

I tweaked the title of the dialog. The Peak Watcher action has also been renamed to clarify that it uses the focus.

I hope this sufficiently addresses the remaining concerns.

TimNoonanVoice commented:
Hi James,

Thanks! The action name and dialog title changes work well here. They address the 'current track' concerns I raised.

Unfortunately, the stray & still shows up on macOS 12.1 for Arm and under Rosetta. 

Thanks
Tim


On 15 Jan 2022, at 9:36 pm, James Teh ***@***.***> wrote:


there is an & at the start of the Follow Track element on Mac: &Follow when last touched track changes unticked checkbox

Is this fixed now?

Or, when it is pressed, could there be an announcement confirming what track is now being watched?

I tweaked the title of the dialog. The Peak Watcher action has also been renamed to clarify that it uses the focus.

I hope this sufficiently addresses the remaining concerns.

—
Reply to this email directly, view it on GitHub <https://github.com/jcsteh/osara/pull/623#issuecomment-1013659134>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGNKY2YMC5UOOPSSGT5MJTDUWFE2TANCNFSM5KWSBPRQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
You are receiving this because you commented.

jcsteh commented:
Ug. I think this stray & character might actually be a bug in the Cockos
GUI library. I'm trying another workaround which should hopefully deal with
this. Can you give it a shot? Thanks.

TimNoonanVoice commented:
Bingo! - Great, the & is there no more.

Thanks for persisting.

Tim

On 16 Jan 2022, at 3:18 pm, James Teh ***@***.***> wrote:


Ug. I think this stray & character might actually be a bug in the Cockos
GUI library. I'm trying another workaround which should hopefully deal with
this. Can you give it a shot? Thanks.
—
Reply to this email directly, view it on GitHub <https://github.com/jcsteh/osara/pull/623#issuecomment-1013807962>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGNKY22FWFISC3WMMTJBJKDUWJBKFANCNFSM5KWSBPRQ>.
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
You are receiving this because you commented.

leonardder commented:
@jcsteh Is there still anything holding this back currently? Would love to see this merged!

jcsteh commented:
Waiting on @ScottChesworth as per https://github.com/jcsteh/osara/pull/623#issuecomment-1005626986

ScottChesworth commented:
Well I reckon we've had all the feedback we're gonna get pre-release now. Ideally I'd like this being merged to be accompanied by an audio demo though. Time is tight here at the moment, but Justin reckons he can get one recorded by middle of next week, so shall we merge then?

RDMurray commented:
> I could potentially have the Watcher dialog report what track it's configuring (e.g. in the title of the dialog), but I worry that would cause a lot of verbosity.

I think it is a good idea to have this information in the dialog.  Could it not just be text in the dialog rather than in the title?  Either way the user could silence speech so the verbosity isn't really a problem.

jcsteh commented:
It's already in the title. I made that change a few weeks back.

RDMurray commented:
Sorry, I didn't notice that.

jcsteh commented:
Ping, two weeks later? I understand an audio demo is ideal, but I also don't want this to languish forever. :)

jcsteh commented:
Note that this now has merge conflicts because of said languishing.

ScottChesworth commented:
@Justinmac84, can we get a progress report? Feel free to email me direct if you still haven't gotten your GitHub posting issue resolved.

ScottChesworth commented:
Unfortunately I didn't get time to make a demo earlier today as I was hoping, and I haven't heard back about any other progress toward one. @jcsteh, feel free to push go, I can be around in the support channels this week to field questions. A demo or stream or whatever can follow. Sorry to have kept it held back for so long. I'll try to plan demos more definitively next time around.
